app/apps/server/bridges/livechat.js

```diff
@@ -48,14 +53,19 @@ export class AppLivechatBridge {
 	async createRoom(visitor, agent, appId) {
 		this.orch.debugLog(`The App ${ appId } is creating a livechat room.`);
 
-		const agentUser = Users.findOneById(agent.id);
-		agentUser.agentId = agentUser._id;
+		let agentRoom;
+		if (agent && agent.id) {
+			const user = Users.getAgentInfo(agent.id);
+			agentRoom = Object.assign({}, { agentId: user._id });
+		}
 
-		return this.orch.getConverters().get('rooms').convertRoom(getRoom({
+		const result = Promise.await(getRoom({
```

Comment:
Considering this method is `async` and is not depending on Meteor's fibers, there is no need to use `Promise.await` and you could instead use `await` itself

Reply:
> Considering this method is `async` and is not depending on Meteor's fibers, there is no need to use `Promise.await` and you could instead use `await` itself

We need to use it because there are other `Promises` inside the `getRoom` method.
Otherwise, for any "throw exception" detected, an error would be raised.
We faced this same case here:

https://github.com/RocketChat/Rocket.Chat/blob/develop/app/livechat/server/api/v1/room.js#L37